### PR TITLE
Add new variable, `BUCKET_NAME`

### PR DESCRIPTION
**Pull request by lorengordon**
This commit adds the `BUCKET_NAME` variable. This variable is used
to support simultaneously both path- and virtualhost- style urls
to the s3 bucket listing, from a single index.html file.
Previously, `BUCKET_URL` could be used to support one method or the
other, but not both methods simultaneously.

### CODE REVIEW

**Comment by lorengordon** [comment]
Here are the various test cases to demonstrate the issue.

| Test | Result | Link |
| --- | --- | --- |
| bucket_url is `null`; access url is virtualhost-based | Success | [link](http://listing-test.s3.amazonaws.com/index-rgrp-null.html) |
| bucket_url is `null`; access url is path-based | Error (Ok, expected) | [link](http://s3.amazonaws.com/listing-test/index-rgrp-null.html) |
| bucket_url is virtualhost-based; access url is virtualhost-based | Success | [link](http://listing-test.s3.amazonaws.com/index-rgrp-vh.html) |
| bucket_url is virtualhost-based; access url is path-based | Error (Fail) | [link](http://s3.amazonaws.com/listing-test/index-rgrp-vh.html) |
| bucket_url is path-based; access url is virtualhost-based | Error (Fail) | [link](http://listing-test.s3.amazonaws.com/index-rgrp-path.html) |
| bucket_url is path-based; access url is path-based | Success | [link](http://s3.amazonaws.com/listing-test/index-rgrp-path.html) |

Basically, the `BUCKET_NAME` variable is intended to address the two failure cases from above.

| Test | Result | Link |
| --- | --- | --- |
| bucket_name is set; access url is virtualhost-based | Success | [link](http://listing-test.s3.amazonaws.com/index-lg-bucketname.html) |
| bucket_name is set; access url is path-based | Success | [link](http://s3.amazonaws.com/listing-test/index-lg-bucketname.html) |

**Comment by rufuspollock** [comment]
Great. OK I'm merging this. One additional request: can we get document for this into the README. It may also be worth including your table somewhere (maybe at bottom of README) or in section about `BUCKET_URL` to make clear what use case it solves.

**Comment by lorengordon** [comment]
Updated the README in https://github.com/rgrp/s3-bucket-listing/pull/42.
